Mage.Sets/src/mage/cards/a/AuroraShifter.java

```diff
@@ -0,0 +1,106 @@
+        super(ownerId, setInfo, new CardType[] { CardType.CREATURE }, "{1}{U}");
+
+        this.subtype.add(SubType.SHAPESHIFTER);
+        this.power = new MageInt(1);
+        this.toughness = new MageInt(3);
+
+        // Whenever Aurora Shifter deals combat damage to a player, you get that many {E}.
+        this.addAbility(new DealsCombatDamageToAPlayerTriggeredAbility(
+                new GetEnergyCountersControllerEffect(SavedDamageValue.MUCH)));
+
+        // At the beginning of combat on your turn, you may pay {E}{E}. When you do, Aurora Shifter becomes a copy of another target creature you control, except it has this ability and "Whenever this creature deals combat damage to a player, you get that many {E}."
+        ReflexiveTriggeredAbility reflexive = new ReflexiveTriggeredAbility(
+                new AuroraShifterCopyEffect(),
+                false);
+        reflexive.addTarget(new TargetPermanent(StaticFilters.FILTER_ANOTHER_TARGET_CREATURE_YOU_CONTROL));
+
+        this.addAbility(new BeginningOfCombatTriggeredAbility(
+                new DoWhenCostPaid(reflexive, new PayEnergyCost(2),
+                        "Pay {E}{E}? When you do, {this} becomes a copy of another target creature you control, except it has this ability and \"Whenever this creature deals combat damage to a player, you get that many {E}.\""),
+                TargetController.YOU, false));
+    }
+
+    private AuroraShifter(final AuroraShifter card) {
+        super(card);
+    }
+
+    @Override
+    public AuroraShifter copy() {
+        return new AuroraShifter(this);
+    }
+}
+
+class AuroraShifterCopyEffect extends OneShotEffect {
+
+    AuroraShifterCopyEffect() {
+        super(Outcome.Copy);
+        this.staticText = "{this} becomes a copy of another target creature you control, except it has this ability and \"Whenever this creature deals combat damage to a player, you get that many {E}.\"";
+    }
+
+    private AuroraShifterCopyEffect(final AuroraShifterCopyEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public AuroraShifterCopyEffect copy() {
+        return new AuroraShifterCopyEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Permanent sourcePermanent = game.getPermanent(source.getSourceId());
+        Permanent copyFromPermanent = game.getPermanent(getTargetPointer().getFirst(game, source));
+
+        if (sourcePermanent != null && copyFromPermanent != null) {
+            game.copyPermanent(Duration.WhileOnBattlefield, copyFromPermanent, sourcePermanent.getId(), source,
+                    new AuroraShifterCopyApplier());
+            return true;
+        }
+        return false;
+    }
+}
+
+class AuroraShifterCopyApplier extends CopyApplier {
+
+    @Override
+    public boolean apply(Game game, MageObject blueprint, Ability source, UUID copyToObjectId) {
+
+        blueprint.getAbilities().add(new DealsCombatDamageToAPlayerTriggeredAbility(
```

Comment:
Don't you need to also add the primary ability? Like Cryptoplasm.

Reply:
No  I don't believe it has the transform ability on the creature it transforms into just the energy gain for damage amount